crates/steel-core/src/env.rs

```diff
@@ -38,15 +38,48 @@ impl Env {
         }
     }
 
-    /// Search starting from the current environment
-    /// for `idx`, looking through the parent chain in order.
-    ///
-    /// if found, return that value
-    ///
-    /// Otherwise, error with `FreeIdentifier`
     #[inline(always)]
     pub fn repl_lookup_idx(&self, idx: usize) -> SteelVal {
-        self.bindings_vec[idx].clone()
+        {
+            let ref this = self.bindings_vec[idx];
+            match this {
```

Comment:
For whatever reason, there was performance improvements by inlining this

Reply:
How does manual inline compare to defining `Clone` with `#[inline]`?

Is it worth looking into [profile guided optimizations](https://doc.rust-lang.org/rustc/profile-guided-optimization.html) or do you think it will add complexity to optimizing?

Reply:
good question, profile guided optimizations is definitely something I should try

I'll check the inline clone as well

Reply:
well that made a huge difference, using `cargo pgo` I shaved like 30% off of the nqueens benchmark

Reply:
Without pgo:
```
λ (nqueen.scm) > (nqueens 13)
=> 73712
Time taken: 32.613855536s
```

With pgo:
```
λ > (nqueens 13)
=> 73712
Time taken: 23.868467826s
```

Reply:
So I think I'll need to create a set of representative input programs for PGO to reliably use it, but overall that is very promising.

Reply:
Confirmed that the clone really don't make much of a difference when using PGO. I don't love it anyway, so for now I just took it out and will look into how to add PGO more robustly when building for distribution.